frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/NotificationHandler.java

```diff
@@ -182,11 +186,18 @@ private String buildContactUsUrl(String referer) {
                 .toString();
     }
 
-    private void writeTestClientOtpToS3(String otp, String destination) {
+    void writeTestClientOtpToS3(NotificationType notificationType, String otp, String destination) {
         Boolean isNotifyDestination =
                 configurationService.getNotifyTestDestinations().contains(destination);
-        if (isNotifyDestination) {
-            LOG.info("Notify Test Destination used in request. Writing to S3 bucket");
+        Boolean isOTPNotificationType =
+                VERIFY_EMAIL.equals(notificationType)
+                        || VERIFY_PHONE_NUMBER.equals(notificationType)
+                        || MFA_SMS.equals(notificationType)
+                        || RESET_PASSWORD_WITH_CODE.equals(notificationType);
```

Comment:
Minor point but something like this might be more readable - 
```suggestion
        var isOTPNotificationType = List.of(VERIFY_EMAIL, MFA_SMS, VERIFY_PHONE_NUMBER, RESET_PASSWORD_WITH_CODE).contains(notificationType);
```